pkg/arm/nameservers.go

```diff
@@ -0,0 +1,40 @@
+package arm
+
+import (
+	"context"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/arm/constants"
+	"github.com/openshift/openshift-azure/pkg/util/azureclient"
+	"github.com/openshift/openshift-azure/pkg/util/azureclient/network"
+)
+
+// GetNameserversFromVnet return the nameservers configured in the vnet or the default
+func GetNameserversFromVnet(ctx context.Context, log *logrus.Entry, subscriptionID, resourceGroup string) ([]string, error) {
+	authorizer, err := azureclient.GetAuthorizerFromContext(ctx, api.ContextKeyClientAuthorizer)
+	if err != nil {
+		return nil, err
+	}
+	vcli := network.NewVirtualNetworkClient(ctx, log, subscriptionID, authorizer)
+	vnet, err := vcli.Get(ctx, resourceGroup, constants.VnetName, "")
+	if err != nil {
+		de, ok := err.(autorest.DetailedError)
+		if ok {
+			re, ok := de.Original.(*azure.RequestError)
+			if ok {
+				if re.ServiceError.Code == "ResourceNotFound" {
```

Comment:
I feel like the ResourceNotFound case here should be avoidable if this is called only on update and not create - can this be done?

Reply:
I'll have a look and see why i needed this